internal/clients/readiness.go

```diff
@@ -118,25 +119,46 @@ func (c DefaultReadinessChecker) checkPendingClient(
 // checkAlreadyExistingClients checks if the already existing clients are still ready to be used and returns the ones
 // that are not.
 func (c DefaultReadinessChecker) checkAlreadyExistingClients(ctx context.Context, alreadyCreatedClients []AlreadyCreatedClient) (turnedPending []adminapi.DiscoveredAdminAPI) {
+	var wg sync.WaitGroup
+	var logMutex sync.Mutex
+	pendingChan := make(chan adminapi.DiscoveredAdminAPI, len(alreadyCreatedClients))
+
 	for _, client := range alreadyCreatedClients {
-		// For ready clients we check readiness by calling the Status endpoint.
-		if ready := c.checkAlreadyCreatedClient(ctx, client); !ready {
-			podRef, ok := client.PodReference()
-			if !ok {
-				// This should never happen, but if it does, we want to log it.
-				c.logger.Error(
-					errors.New("missing pod reference"),
-					"Failed to get PodReference for client",
-					"address", client.BaseRootURL(),
-				)
-				continue
+		wg.Add(1)
+		go func(client AlreadyCreatedClient) {
+			defer wg.Done()
+
+			// For ready clients we check readiness by calling the Status endpoint.
+			if ready := c.checkAlreadyCreatedClient(ctx, client); !ready {
+				podRef, ok := client.PodReference()
+				if !ok {
+					logMutex.Lock()
```

Comment:
Do we need to use the mutex to guard the logger? I'm not sure we've ever used that anywhere else. Any docs mention that we need to use this?

Reply:
Sorry for the delayed reply. I have checked the log library we are using, and based on relevant records such as code and issues, it is indeed unnecessary to add a lock. I will update this PR accordingly.